### DISCUSSION TITLE
I2C/BH1750 problem

### DISCUSSION THREAD

**Discussion by andrewbpowers** [arendst/Tasmota#12615] [Q&A] [answered]
Hello!

I try to connect a BH1750 sensor to a ESP-01S with a Tasmota Sensor bin. 
I would say I have already tried all possible combinations GPIO<->SDA/SCL. 
I have also tried both addresses (0x23/0x5C ) for the sensor. 
A DHT22 on GPIO 3 for example, is recognized immediately.

What might I have missed?
Thanks in advance for any help and tips!

Take care...

**Comment by barbudor** [comment, marked as answer]
Are you using tasmota-sensors ?
Always check [BUILDS.md](https://github.com/arendst/Tasmota/blob/development/BUILDS.md) to be sure if a sensor or feature is included in the binary that you are using.

This being said, GPIO0 and GPIO2 should be ok for I2C as they should have on-board pull-ups on the ESP01S and BH1750 shouldn't disturb the boot of the ESP.

**Reply by sfromis** [reply]
Output of the command I2CScan should include address 0x23 (or 0x5C), and I2CDriver output should include driver number 11. Until that is the case, no chance of Tasmota doing anything.

**Comment by andrewbpowers** [comment]
@barbudor 
Thank you for your feedback! :-)

Yes, I use "tasmota.bin" (http://ota.tasmota.com/tasmota/release/tasmota.bin).
I see the BH1750 in the table ("USE_BH1750") and that other people have also already used the sensor with Tasmota.
I would like to say GPIO0/GPIO2 could be sometime criticals, especially during the boot process. With a DHT22 on the GPIO2 from a ESP-01S for example, it doesn't works. But the DHT11 works fine. 
I try the combination GPIO0 and GPIO2 as well, but the BH1750 was also not detected. :-(

**Comment by andrewbpowers** [comment]
@sfromis 
Thanks for the tip with "I2CScan"... I didn't know it yet! ;-)
I think I'll keep trying with it for now and see if I can locate the error with it.

**Reply by barbudor** [reply]
The error is in your answer : you are using tasmota and not tasmota-sensors

![image](https://user-images.githubusercontent.com/2996042/125171874-ed82c580-e1b6-11eb-8b30-7b978d903bc1.png)

**Comment by andrewbpowers** [comment]
@barbudor 
I was too quick in my answer. Sorry!
I tried the "normal" version first, then the sensor version (http://ota.tasmota.com/tasmota/release/tasmota-sensors.bin). But it didn't work with either of them!

Just wrote a small test program for the I2C bus and connected the BH1750 to a NodeMCU. 
With this, the sensor was not found either. Then I checked all the pins on the sensor and apparently the problem was a cold solder joint. No idea why, but the sensor has been in a box for a long time and has never been used. Pretty stupid error! 
Anyway, both my test program and Tasmota detected the sensor immediately. 

Thanks again for your quick help and support!
